Disable 'SVG 1.1' deprecated properties behind feature flag

Ahmad-S792 opened the pull request:
#### a229ccbd2e800bfbaee55c28c40aa3f3b7cb22c3
<pre>
Disable &apos;SVG 1.1&apos; deprecated properties behind feature flag
<a href="https://bugs.webkit.org/show_bug.cgi?id=295261">https://bugs.webkit.org/show_bug.cgi?id=295261</a>
<a href="https://rdar.apple.com/154733488">rdar://154733488</a>

Reviewed by NOBODY (OOPS!).

This patch adds new flag called &apos;EnableSVGDeprecatedProperties&apos;, which
gives access to deprecated SVG properties listed below, which are removed
from SVG 2 specification and also removed from Gecko / Firefox and
Blink / Chrome.

- nearestViewportElement [Removed from Gecko / Firefox]
- farthestViewportElement [Removed from Gecko / Firefox]
- useCurrentView [Removed from Gecko / Firefox &amp; Blink / Chromium ]
- currentView [Removed from Gecko / Firefox &amp; Blink / Chromium ]

This flag will be enabled in Safari Technology Preview to obtain data
whether the following can be removed from WebKit without any web
compatibility issues.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/svg/SVGGraphicsElement.idl:
* Source/WebCore/svg/SVGSVGElement.idl:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a229ccbd2e800bfbaee55c28c40aa3f3b7cb22c3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/109829 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29487 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19917 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/115850 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60066 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/111792 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30165 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38075 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83463 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/112777 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24046 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/98904 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63926 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23425 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17053 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/59645 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/102320 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93416 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17096 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/118642 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/108381 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/36868 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27320 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92474 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37241 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95172 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92297 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37263 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15007 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/32714 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/36763 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42233 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/132657 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36423 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35907 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/39765 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38132 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/97b8d030435c78d79381b648c11caffed6e7882b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47426

Ahmad-S792 commented:
> nit:
> 
> ```diff
> - Disable `SVG 1.1` historical properties behind featre flag
> + Disable `SVG 1.1` historical properties behind feature flag
> ```
> 
> in the commit message

It is fine for now since I want to see EWS results first.

Ahmad-S792 committed:
Disable 'SVG 1.1' deprecated properties behind feature flag
https://bugs.webkit.org/show_bug.cgi?id=295261
rdar://154733488

Reviewed by NOBODY (OOPS!).

This patch adds new flag called 'EnableSVGDeprecatedProperties', which
gives access to deprecated SVG properties listed below, which are removed
from SVG 2 specification and also removed from Gecko / Firefox and
Blink / Chrome.

- nearestViewportElement [Removed from Gecko / Firefox]
- farthestViewportElement [Removed from Gecko / Firefox]
- useCurrentView [Removed from Gecko / Firefox & Blink / Chromium ]
- currentView [Removed from Gecko / Firefox & Blink / Chromium ]

This flag will be enabled in Safari Technology Preview to obtain data
whether the following can be removed from WebKit without any web
compatibility issues.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/svg/SVGGraphicsElement.idl:
* Source/WebCore/svg/SVGSVGElement.idl:

annevk commented:
I think preview is also enabled for testing, so I doubt you'll see any fallout. Though maybe on the Linux bots?

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a229ccbd2e800bfbaee55c28c40aa3f3b7cb22c3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47426

nt1m commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -2635,6 +2635,20 @@ EnableInheritURIQueryComponent:
 
+EnableSVGDeprecatedProperties:
+  type: bool
+  status: preview
```
I think the preference is doing the opposite of what you want here.

By default, the deprecated properties would be disabled, and in STP the deprecated properties would be enabled.

@brentfulgham can you confirm this is how it would work?

Ahmad-S792 replied:
This is my second attempt to add preferences, so I am expecting to mess-up things but it is also learning opportunity. :-)

brentfulgham replied:
We should use `status: stable` here. Otherwise, STP will turn this feature on for anyone launching the app.

I assume the goal is to have the deprecated properties off by default when you use Safari (just like they are in Firefox and Chrome). Using `stable` will put the flag in the right place, but allow the user to turn the flag back on if they need it.

brentfulgham reviewed (changes requested):
Please change to `stable` status, and this is ready to go!
